src/assets/demo/validateInput.js

```diff
@@ -134,3 +134,26 @@ function logNativeFocusIn(event) {
 function logNativeFocusOut(event) {
   console.log("native focusout", event.target.id);
 }
+
+function enableEditingForInput(id) {
+  const input = document.querySelector(id);
+  input.addEventListener('calciteInputEnableEditing', () => input.editingEnabled = true)
+}
+
+function saveChangesForInput(id, isSuccess = true) {
+  const input = document.querySelector(id)
+  const action = input.querySelector('[slot="input-action"]');
+  const label = document.querySelector(`${id}-status`);
+  action.addEventListener('click', () => {
```

Comment:
I wonder if the "saving changes with action" should be handled by the input component, and then when acted upon emit an event that the user has requested a "save / update", so this doesn't have to be set up like this?

Maybe an 'editable-type="inputblur | buttonclick"' (bad nomenclature but...), and then the input component would render that action for users?

Reply:
Similarly, we shouldn't rely on the dev to set "clearable" on the input to show the clear button, I think if the dev opts in to "show a button to save / emit vs. emit on blur", we should internally set clearable so the clear button / Escape key is always there for a user (not relevant on the version that emits on blur)

Reply:
If I'm understanding correctly, the `clearable` prop is only required for non inline-editable inputs. The clear button is added automatically if the input is inline editable and has been passed a child action via the action slot.

Reply:
Re: the save action, doesn't this approach keep the API more consistent with the non inline-editable inputs? Any saving logic is handled directly by `onClick` handler on that slotted button and that the consumer can affect the input styles accordingly.